src/app/profile/spaces/space.service.ts

```diff
@@ -0,0 +1,100 @@
+import { Injectable } from '@angular/core';
+import { Headers, Http } from '@angular/http';
+import { cloneDeep } from 'lodash';
+import { Logger } from '../../shared/logger.service';
+import { AuthenticationService } from '../../auth/authentication.service';
+import { Space } from '../../models/space';
+
+@Injectable()
+export class SpaceService {
+
+  private headers = new Headers({ 'Content-Type': 'application/json' });
+  private spacesUrl = process.env.API_URL + "spaces";
+  private nextLink: string = null;
+
+  private spaces: Space[] = [];
+  private spaceIdIndexMap = {};
+
+  constructor(private http: Http, private logger: Logger, private auth: AuthenticationService) {
+    if (this.auth.getToken() != null) {
+      this.headers.set('Authorization', 'Bearer ' + this.auth.getToken());
+    }
```

Comment:
I looked at this comment, but put this on the shelf until further discussion. IMHO, the cleanest practice suitable for us, is to use a separate http service that handles bearer token auth, and delegates to the Angular http client service for other actions, like so: http://stackoverflow.com/a/34465070/3916